src/library/librarybackend.cpp

```diff
@@ -218,11 +218,11 @@ void LibraryBackend::UpdateTotalSongCount() {
 }
 
 void LibraryBackend::AddDirectory(const QString& path) {
-  QString canonical_path = QFileInfo(path).canonicalFilePath();
-  QString db_path = canonical_path;
+  QString absolute_path = QFileInfo(path).absoluteFilePath();
+  QString db_path = absolute_path;
 
   if (Application::kIsPortable && Utilities::UrlOnSameDriveAsClementine(
-                                      QUrl::fromLocalFile(canonical_path))) {
+                                      QUrl::fromLocalFile(absolute_path))) {
```

Comment:
Shouldn't we resolve symlinks for this check?

Reply:
This will only return false on Windows.

IMO, Unix or otherwise, symlinks are not of implicit occurance, and the user ought to already have a reasonable awareness of consequence. Thus, Clementine should only malfunction when the user (or the OS) does, and have no capability for or awareness beyond that.

Reply:
Resolve this only for Windows within UrlOnSameDriveAsClementine().